js/src/name/publish.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-env mocha */
+'use strict'
+
+const hat = require('hat')
+
+const { fixture } = require('./utils')
+const { spawnNodeWithId } = require('../utils/spawn')
+const { getDescribe, getIt, expect } = require('../utils/mocha')
+
+module.exports = (createCommon, options) => {
+  const describe = getDescribe(options)
+  const it = getIt(options)
+  const common = createCommon()
+
+  describe('.name.publish', function () {
+    const keyName = hat()
+    let ipfs
+    let nodeId
+
+    before(function (done) {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+
+        spawnNodeWithId(factory, (err, node) => {
+          expect(err).to.not.exist()
+
+          ipfs = node
+          nodeId = node.peerId.id
+
+          ipfs.files.add(fixture.data, { pin: false }, done)
+        })
+      })
+    })
+
+    after((done) => common.teardown(done))
+
+    it('should publish an IPNS record with the default params', (done) => {
+      this.timeout(50 * 1000)
```

Comment:
Apologies I've just noticed that a bunch of these tests are using `this.timeout` in an arrow function - you'll need to change it to `function (done) {/*...*/}` for it to be referring to the correct `this` (otherwise you're referring to `this` from the `describe` block)